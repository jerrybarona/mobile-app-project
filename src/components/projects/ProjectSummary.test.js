import React from "react";
import ProjectSummary from "./ProjectSummary";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const firebase = require("firebase");

const mockProject = {
  id: 1,
  title: "foo",
  authorFirstName: "John",
  authorLastName: "Doe",
  createdAt: new firebase.firestore.Timestamp(1554972151, 191000000)
};

function renderProjectSummary(args) {
  const defaultProps = {
    project: mockProject
  };

  const props = { ...defaultProps, ...args };
  return shallow(<ProjectSummary {...props} />);
}

it("Should render ProjectSummary component and match snapshot", () => {
  const tree = renderer.create(<ProjectSummary project={mockProject} />);
  expect(tree).toMatchSnapshot();
});

it("Should find ProjectSummary component title", () => {
  const wrapper = renderProjectSummary();
  expect(wrapper.find("span").text()).toEqual(mockProject.title);
});
