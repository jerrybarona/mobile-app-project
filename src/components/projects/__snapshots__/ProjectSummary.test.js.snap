// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProjectSummary component and match snapshot 1`] = `
<div
  className="card z-depth-0 project-summary"
>
  <div
    className="card-content grey-text text-darken-3"
  >
    <span
      className="card-title "
    >
      foo
    </span>
    <p>
      Posted by 
      John
       
      Doe
    </p>
    <p
      className="grey-text"
    >
      04/11/2019
    </p>
  </div>
</div>
`;
